### PR TITLE
Implement profile privacy and security page

### PR DESCRIPTION
**Pull request by TanyaaCJain**
Should merge after PR #8 is merged.

### CODE REVIEW

**Comment by miss-queenNL** [comment]
- arrow down when closed and up when open. (manage assets, analyse, flights, 
- Settings : make it Account settings or business settings if it is a business account
- missing the multiple pages for settings and billing
- Padding within notifications and myprofile should be the same as the menu itself (the top of the popup does seem to have the right padding)
- hovering the my profile arrow doesn't show a different mouse
- clicking the entire area of "my profile" and "artic ocean" should open up the functionality behind it (just like the accordion responds when you click on the text rather than the arrow)
- Notification close "X" does not have a mouse over icon
- My profile close "X" does not have a mouse over icon
- Arctic ocean close "X" does not have a mouse over icon
- Arrow of "My profile" should be up (follow the movement of the popup) and down for closing
- Clicking My profile when the popup is opened should close the popup again (rather than go to the profile settings page)
- missing the dark overlay over the menu when a pop-up is active. Also expecting the dark overlay to be clickable (and if clicked close the popup)
- The accordion gets rounded corners when selected (see flights)
image
- Switching between accordion items does not close the previous one open. (there can be 2 accordions open). Would prefer if one is open and then the next is opened, that the first closes again
- payment may be deleted from the my profile (will move to billing or settings)


Question if we change this:
- Clicking a different company in the "switch account" popup should just load the dashboard again (since we will use this as a demo) As long as this functionality is not developped yet.  (or for now have no functionality at all untill we develop it?)
- current company selected should not be clickable (or for now have no functionality at all untill we develop it?)
- Clicking artic ocean (if it is the current selected company) should close the popup. If a different company is selected this area should change into the new name and the button should become orange

Note:
I'm calling pop-up the area's that overlay the menu, it may not actually be pop-ups.

File: src/components/profile/privacy-and-security/password-modal.vue [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+const changePassword = () => {
+    console.log('change password');
+};
+const resetData = () => {
+    verificationCode.value = '';
+    newPass.value = '';
+    confirmPass.value = '';
+    step.value = 1;
+};
+</script>
```
Comment (Mozes721) [inline src/components/profile/privacy-and-security/password-modal.vue]: could you organize it by ref, methods, object plus with empty line to make it clearer or comment to define it?

```
<script setup>
  import { PhPencil, PhUserCircle, PhX } from "@phosphor-icons/vue";
  import InputText from '@/components/input-fields/input-text.vue';
  import InputPassword from "@/components/input-fields/input-password.vue";
 // Ref Object Data
  const verificationCode = ref('');
  const newPass = ref('');
  const confirmPass = ref('');
  let step = ref(1);
  const open = ref(false);

  // Methods/Functions
  const sendVerificationCode = () => {
    step.value = 2;
  };

  const showModal = () => {
    open.value = true;
    resetData();
  };

  const closeModal = () => {
    open.value = false;
  };

  const submit = () => {
    if (!verificationCode.value || !newPass.value || !confirmPass.value) {
      console.log('all fields must be filled');
      return;
    }
    if (newPass.value !== confirmPass.value) {
      console.log('passwords do not match');
      return;
    }
    changePassword();
    closeModal();
  };

  const changePassword = () => {
    console.log('change password');
  };
 // Object Data
  const resetData = () => {
    verificationCode.value = '';
    newPass.value = '';
    confirmPass.value = '';
    step.value = 1;
  };
</script>
```

File: src/pages/profile/privacy-and-security.vue [hunk 2]
```diff
@@ -2,11 +2,69 @@
...
+    "opt ins": [
+        { name: 'Receive all FLYARROW newsletters', checked: false },
+        { name: 'Receive discounts and promotions', checked: true },
+    ],
+    "email alerts": [
+        { name: 'All', checked: true },
+        { name: 'Receive message when new team member is added', checked: true },
+        { name: 'Receive message when vehicles are added/changed', checked: false },
+    ],
+});
```
Comment (Mozes721) [inline src/pages/profile/privacy-and-security.vue:63]: This i noticed in other PR so there is duplication maybe invoke it from service dir if it encapsulates business logic across or utils if it's generic-purpose utility function or helper

**Review by owlot — CHANGES_REQUESTED** [review]
Can you check and address the issues listed by @miss-queenNL and the comments made by @Mozes721 ?

File: src/pages/profile/privacy-and-security.vue (hunk 2 above)
Comment (TanyaaCJain) [reply src/pages/profile/privacy-and-security.vue:63]: Can you please share the link for this? I am not aware of any such redundancy. I believe a new issue would be better for project wide changes.

File: src/components/profile/privacy-and-security/password-modal.vue (hunk 1 above)
Comment (TanyaaCJain) [reply src/components/profile/privacy-and-security/password-modal.vue]: Yes, documenting always is a good idea. It can be looked into

**Comment by TanyaaCJain** [comment]
Thank you @miss-queenNL for the review. It seems the issues are not related to this task, rather the side navigation. Can you please open a new issue for this? I would then look into the concerns, discuss and work accordingly. 

@owlot Do you think a new dework task would be preferable?

File: src/pages/profile/privacy-and-security.vue (hunk 2 above)
Comment (Mozes721) [reply src/pages/profile/privacy-and-security.vue:63]: mhm ok never mind as in other PR it's at 
```
PR dw-1239/payment
src/pages/profile/privacy-and-security.vue 
line 49
```
But I guess would be better to add new code just in one PR and other would inherit it after merge, but can just pass it for now.

File: src/components/profile/privacy-and-security/password-modal.vue (hunk 1 above)
Comment (TanyaaCJain) [reply src/components/profile/privacy-and-security/password-modal.vue]: @Mozes721 I was going over this, and if you look closely reactive variables were defined first and then methods/functions. Though the comments can be added, it doesn't seem to need the definition.

**Review by owlot — APPROVED** [review]

**Pull request merged**